cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -177,15 +177,16 @@ runQueryProtocolParametersCmd
           . newExceptT $ executeQueryAnyMode localNodeConnInfo qInMode
   writeProtocolParameters sbe mOutFile pp
   where
-    -- TODO: Conway era - use ledger PParams JSON
     writeProtocolParameters
       :: ShelleyBasedEra era
       -> Maybe (File () Out)
       -> L.PParams (ShelleyLedgerEra era)
       -> ExceptT QueryCmdError IO ()
     writeProtocolParameters sbe mOutFile' pparams =
       firstExceptT QueryCmdWriteFileError . newExceptT
-        $ writeLazyByteStringOutput mOutFile' $ encodePretty $ fromLedgerPParams sbe pparams
+        $ writeLazyByteStringOutput mOutFile'
+          $ shelleyBasedEraConstraints sbe
+            $ encodePretty pparams
```

Comment:
What about this TODO: https://github.com/IntersectMBO/cardano-cli/blob/main/cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs#L506
Are we able to round-trip i.e. read the parameters into the `transaction build` after your change? Afair there was a testnet test for that.

Reply:
Notes to myself:

* On one hand, `transaction build` doesn't read the parameters by using the CLI's code: it requests them from the node
* On the other hand, `transaction build-raw` does read parameters from the command line.

Reply:
@carbolymer> actually there are multiple usages of the API JSON encoding:

1. In `build-estimate`: https://github.com/IntersectMBO/cardano-cli/blob/main/cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs#L286
2. In `build-raw` (the one you pointed out)
3. In `calculate-min-fee`: https://github.com/IntersectMBO/cardano-cli/blob/main/cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs#L1261
4. In `calculate-min-value`: https://github.com/IntersectMBO/cardano-cli/blob/main/cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs#L1342

Should we change all of them?

@gitmachtl> would you be able to test this change before we merge it? I'm afraid we are not the best ones to test this change exhaustively.

Reply:
sure, also those are really important. let me compile the cli for testing...

Reply:
@smelc so first issue i see is, that cli is complaining about the missing parameter `key "minUTxOValue" not found`

so the parameters with `null` value that were sorted out are causing an issue here. cli expects at least that the key is in there.

Reply:
@gitmachtl> is there a way I can reproduce your tests myself? So that then I can investigate how to fix the issues in autonomy.

Reply:
@smelc yes, we should only use one encoding, so in this case ledger's.